clcache.py

```diff
@@ -74,6 +80,80 @@ def __str__(self):
         return repr(self.message)
 
 
+class ManifestsManager(object):
+    HASH_INVALIDATION_COUNTER = 3
+
+    def __init__(self, manifestsRootDir):
+        self._manifestsRootDir = manifestsRootDir
+
+    def manifestDir(self, manifestHash):
+        return os.path.join(self._manifestsRootDir, manifestHash[:2])
+
+    def manifestPath(self, manifestHash):
+        return os.path.join(self.manifestDir(manifestHash), manifestHash + ".json")
+
+    def setManifest(self, manifestHash, manifest):
+        ensureDirectoryExists(self.manifestDir(manifestHash))
+        with open(self.manifestPath(manifestHash), 'w') as outFile:
+            # Convert namedtupel to dict to get key names
+            doc = {
+                'includeFiles': manifest.includeFiles,
+                'includesContentToObjectMap': manifest.includesContentToObjectMap
+            }
+            json.dump(doc, outFile, sort_keys=True, indent=2)
```

Comment:
There is a [`collections.namedtuple._asdict`](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict) method to convert a `namedtuple` to a dictionary.

A [StackOverflow answer](http://stackoverflow.com/a/26180604/91757) explains:

> This is a documented method of namedtuples, i.e. unlike the usual convention in python the leading underscore on the method name isn't there to discourage use. Along with the other methods added to namedtuples, _make, _replace, _source, _fields, it has the underscore only to try and prevent conflicts with possible field names.